src/tcgwars/logic/impl/gen3/RubySapphireNG.groovy

```diff
@@ -294,14 +323,24 @@ public enum RubySapphireNG implements LogicCardInfo {
         pokePower "Energy Draw", {
           text "Once during your turn (before your attack), you may discard 1 Energy card from your hand. Then draw up to 3 cards from your deck. This power can't be used if Delcatty is affected by a Special Condition."
           actionA {
+            checkNoSPC()
+            checkLastTurn()
+            assert my.hand.filterByType(ENERGY) : "No Energy in hand"
+            //Using LV.X Compendium Ruling instead of the EX: You can discard even if you don't have cards in deck.
+            powerUsed()
+
+            my.hand.filterByType(ENERGY).select("Discard").discard()
+            if (my.deck){
+              def maxDraw = Math.min(3, my.deck.size())
+              draw choose(1..maxDraw, "Draw how many cards?")
+            }
```

Comment:
Don't remember the details (would need to check the git blame for PK, from before moving the card over here), but many of these cards need to do this check to avoid a NPE iirc.

Also, there's still the need in most EX-era cards to be able to select "up to x cards" to draw, unlike modern cards which usually force the max amount (or as much as possible).

Reply:
If `draw 3` causes an NPE, it's probably better to fix that in `DrawCard` instead of having to do this logic on literally every card that draws cards. However, `DrawCard` does check if a Deck is already size 0 and skips drawing from it in that case from what I see.

Reply:
I did miss that this was an "up to" implementation. Maybe we could add this as a static `drawUpTo` instead?

Reply:
Sounds like a great idea. Separate method, or integrated into the draw static itself?

In any case, probably worth its own PR (will add a couple of these to trello in a bit)

Reply:
Best to be it's own method I think. Easier to read the method name `drawUpTo` instead of having to read (and create) javadoc for a method that seems simple enough currently.